internal/controller/resourcesetinputprovider_controller_test.go

```diff
@@ -21,6 +21,182 @@ import (
 	fluxcdv1 "github.com/controlplaneio-fluxcd/flux-operator/api/v1"
 )
 
+func TestResourceSetInputProviderReconciler_GitLabBranch_LifeCycle(t *testing.T) {
```

Comment:
It'd be nice to cover GitHubBranch and GitLabMergeRequest life-cycles as well.

Reply:
We do have tests for these in the provider package. Since the exported inputs are the same, the coverage will not increase for the reconciler.